File: cliente/src/main/java/hct/huaripaucar/jesus/cliente/service/ClienteService.java
package hct.huaripaucar.jesus.cliente.service;

import hct.huaripaucar.jesus.cliente.model.Cliente;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface ClienteService {

    Flux<Cliente> listar();

    Mono<Cliente> obtener(Long id);

    Mono<Cliente> crear(Cliente cliente);

    Mono<Cliente> actualizar(Long id, Cliente cambios);

    Mono<Cliente> eliminar(Long id);

    Mono<Cliente> restaurar(Long id);
}

File: frontend-hackaton/src/app/core/services/cliente.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Cliente } from '../interfaces/cliente.interface';

@Injectable({ providedIn: 'root' })
export class ClienteService {

  private baseUrl = 'https://fuzzy-happiness-69rx57jv4xv534x7v-30082.app.github.dev/api/clientes';

  constructor(private http: HttpClient) {}

  listar(): Observable<Cliente[]> {
    return this.http.get<Cliente[]>(this.baseUrl);
  }

  crear(c: Cliente): Observable<Cliente> {
    return this.http.post<Cliente>(this.baseUrl, c);
  }

  actualizar(id: number, c: Cliente): Observable<Cliente> {
    return this.http.put<Cliente>(`${this.baseUrl}/${id}`, c);
  }

  eliminar(id: number): Observable<Cliente> {
    return this.http.patch<Cliente>(`${this.baseUrl}/${id}/eliminar`, {});
  }

  restaurar(id: number): Observable<Cliente> {
    return this.http.patch<Cliente>(`${this.baseUrl}/${id}/restaurar`, {});
  }
}